### ISSUE TITLE
ML-Agent loss jump 10000% after long train

### ISSUE THREAD

**Issue by thammachet** [Unity-Technologies/ml-agents#5550]
I have trained the turtle to find the way to water using a "camera sensor" which feeds 84*84 pixels to NN. It was getting smarter and better. Then at some point, the loss jump and reward reduce dramatically. The turtle becomes crazy in a few epochs. What causes this problem? Is there a bug in gradient? How to prevent this thing from happening again?

Link to project
https://drive.google.com/file/d/1T7s_V8sudtKrjGorFzV9kB8ONe2ak48_/view?usp=sharing
Config
In zip file -> Config -> TurtleAgent1.yaml

1. Train for long time.
2. And use Big NN.

- Unity Version: 2020.3.18f1
- OS + version: Windows 10
- _ML-Agents version_: 2.1.0-exp.1
- _Torch version_: 1.7.1+cu110

This screenshot only show latest experienced bug It occur every time.
![Screenshot 2021-09-25 160153](https://user-images.githubusercontent.com/11439607/134847491-0897770c-856b-48da-916d-b0116b9d39ef.png)
![Screenshot 2021-09-27 113503](https://user-images.githubusercontent.com/11439607/134847518-3aa98629-b7f9-4d1f-84d1-d888829fa091.png)
![Screenshot 2021-09-27 113630](https://user-images.githubusercontent.com/11439607/134847524-10cf2961-b1b6-4304-b132-509842bb8523.png)
![Screenshot 2021-09-27 113751](https://user-images.githubusercontent.com/11439607/134847531-00d9b023-d4b4-4a9a-b779-eecc1d03b434.png)
![Screenshot 2021-09-27 113804](https://user-images.githubusercontent.com/11439607/134847501-fa5f2358-6f04-4fb0-a3b8-978891a6067e.png)

**Comment by thammachet** [comment]
It seem like decrease learning rate will help with this problem. I will close issue for now. Thank you.

**Comment by thammachet** [comment]
After more experimentation, I found that a decrease learning rate not solving this issue. It just slows the event waiting to happen. After some research, the possible cause might be a gradient explosion? 
Here are some more virtual evidence for this problem.

https://www.youtube.com/watch?v=BHb_-0QGQUU
The problem occurs around 7:01:27
The turtle becomes brainless in a second.

**Comment by hvpeteet** [comment]
Hi, thanks for your interest in ml-agents. This issue looks like an issue with how training is set up (a hard problem, but not one we can fix on GitHub). For help with a specific environment please post on our forums https://forum.unity.com/forums/ml-agents.453/.
